src/shared/components/lesson-card-manager/lesson-card-manager.component.ts

```diff
@@ -0,0 +1,258 @@
+                .filter(Boolean)
+                .map(formatSchedule)
+                .filter(Boolean)
+            : [];
+
+          return {
+            classId: requestId,
+            userName: studentData.username,
+            educationLevel: studentData.educationLevel.levelType,
+            subject:
+              request.subject.subjectName || 'Erro ao obter a disciplina',
+            reasonType: request.reason
+              .map((r: string) => ReasonLabel[r as Reason] || r)
+              .join(', '),
+            tutorDescription:
+              request.additionalInfo || 'Sem informações adicionais',
+            availableSchedules: this.getAvailableSchedules(formattedSchedules),
+            subjectId: request.subject.subjectId,
+            studentId: studentData.id,
+          };
+        });
+      }
+    } catch (error) {
+      console.error('Erro ao carregar dados do usuário', error);
+      this.errorMessage = 'Erro ao carregar dados do usuário';
+    }
+  }
+
+  /**
+   * Manipula o clique no botão de edição.
+   * @param request - Solicitação a ser editada.
+   */
+  onEditClick(request: any): void {
+    if (!request.classId) {
+      console.error('Tentativa de edição sem ID da solicitação:', request);
+      return;
+    }
+
+    // Fechar qualquer modal aberto anteriormente
+    if (this.showEditForm) {
+      this.showEditForm = false;
+      this.selectedRequest = null;
+      // Dar tempo para o componente ser destruído adequadamente
+      setTimeout(() => {
+        this.openEditForm(request);
+      }, 100);
+    } else {
+      this.openEditForm(request);
+    }
+  }
+
+  /**
+   * Abre o formulário de edição.
+   * @param request - Solicitação a ser editada.
+   */
+  private openEditForm(request: any): void {
+    try {
+      if (!request) {
+        console.error('Request inválido:', request);
+        return;
+      }
+
+      // Garantir que todos os campos necessários existam
+      this.selectedRequest = {
+        classId: request.classId,
+        userName: request.userName || 'Usuário não identificado',
+        educationLevel: request.educationLevel || 'Nível não definido',
+        subject: request.subject || 'Sem matéria definida',
+        reasonType: request.reasonType || '',
+        tutorDescription: request.tutorDescription || '',
+        availableSchedules: Array.isArray(request.availableSchedules)
+          ? request.availableSchedules.filter(Boolean)
+          : [],
+        subjectId: request.subjectId,
+      };
+
+      this.showEditForm = true;
+      this.cdr.detectChanges();
+    } catch (error) {
+      console.error('Erro ao preparar dados para edição:', error);
+    }
+  }
+
+  /**
+   * Fecha o formulário de edição.
+   * @param result - Resultado da edição.
+   */
+  onCloseForm(result?: string): void {
+    this.showEditForm = false;
+    this.selectedRequest = null;
+    this.cdr.detectChanges();
+
+    if (result === 'updated') {
+      setTimeout(() => {
```

Comment:
Em vez de timeouts
    timer(100).pipe(take(1)).subscribe(() => {
      this.openEditForm(request);
    });